samples/setEndpoint.js

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+async function setEndpoint() {
+  // [START vision_set_endpoint]
```

Comment:
This is a tiny bit of, the `START` should be above the `sendEndpoint` method, so that the example code includes both the start of the function and the closing `}`.

The sample guidelines also suggest that we have an outer function that handles command line arguments, and then an inner `async` function, see:

```js
function main(arg1 = 'default_value_one', arg2 = 1234) {
  // [START your_region_tag]
  /**
   * TODO(developer): Uncomment these variables before running the sample.
   */
  // const arg1 = 'default_value_one';
  // const arg2 = 1234;

  // Imports the Google Cloud Some API library
  const {SomeApiClient} = require('@google-cloud/some-api');

  // Instantiates a client
  const someApiClient = new SomeApiClient();

  async function doSomething() {
    ...
  }

  doSomething();
  // [END your_region_tag]
}

main(...process.argv.slice(2));
```

Reply:
The intention of the sample is to only highlight that small section of the function, the rest of the function is being left so that the tests still pass as this is a modified version of the quickstart. Sorry, should have mentioned this with the PR. 

Here's the Python canonical (within the tags): https://github.com/GoogleCloudPlatform/python-docs-samples/pull/2497/commits/750d2f6d36b1a030f7226067922c8ac3dab3a39b#diff-84edde0650e4e711244c1abf96593fc3

Reply:
gotcha, what's weird to me, is I think this includes the closing brace, so you end up with:

```js
  // Imports the Google Cloud client library
  const vision = require('@google-cloud/vision');

  // Specifies the location of the api endpoint
  const clientOptions = {apiEndpoint: 'eu-vision.googleapis.com'};

  // Creates a client
  const client = new vision.ImageAnnotatorClient(clientOptions);

  // Performs label detection on the image file
  const [result] = await client.labelDetection('./resources/wakeupcat.jpg');
  const labels = result.labelAnnotations;
  console.log('Labels:');
  labels.forEach(label => console.log(label.description));
}
```

where the last `}` is the closing brace for `async function setEndpoint() {` which is out of context for the algorithm; does the END just need to be brought up a couple lines?

Reply:
Ahh yes! Good catch, it should certainly be a few lines higher. Fixed :)